software/deploy/deploy.go

```diff
@@ -175,26 +161,72 @@ func buildPushDockerImage(houstonClient houston.ClientInterface, c *config.Conte
 			os.Exit(1)
 		}
 	}
-	imageName := airflow.ImageName(name, "latest")
+	return nil
+}
 
+func buildPushDockerImage(houstonClient houston.ClientInterface, c *config.Context, deploymentInfo *houston.Deployment, name, path, nextTag, cloudDomain, byoRegistryDomain string, ignoreCacheDeploy, byoRegistryEnabled bool, description, customImageName string) error {
+	imageName := airflow.ImageName(name, "latest")
 	imageHandler := imageHandlerInit(imageName)
-
 	if description != "" {
 		deployLabels = append(deployLabels, "io.astronomer.deploy.revision.description="+description)
 	}
 
-	buildConfig := types.ImageBuildConfig{
-		Path:            config.WorkingPath,
-		NoCache:         ignoreCacheDeploy,
-		TargetPlatforms: deployImagePlatformSupport,
-		Output:          true,
-		Labels:          deployLabels,
-	}
+	var err error
+	if customImageName == "" {
```

Comment:
nit: can we simplify this function logic it seems to be having too many independent if-else logic, we could probably break it down into two-three independent functions -> `buildImage`, `validateCustomImage`, `pushImage` probably